x-pack/filebeat/input/httpjson/rate_limiter.go

```diff
@@ -12,34 +12,49 @@ import (
 	"strconv"
 	"time"
 
+	"golang.org/x/time/rate"
+
 	"github.com/elastic/elastic-agent-libs/logp"
 )
 
 type rateLimiter struct {
 	log *logp.Logger
 
-	limit      *valueTpl
-	reset      *valueTpl
-	remaining  *valueTpl
-	earlyLimit *float64
+	limit         *valueTpl
+	reset         *valueTpl
+	remaining     *valueTpl
+	earlyLimit    *float64
+	clientLimiter *rate.Limiter
 }
 
 func newRateLimiterFromConfig(config *rateLimitConfig, log *logp.Logger) *rateLimiter {
 	if config == nil {
 		return nil
 	}
 
-	return &rateLimiter{
+	limiter := &rateLimiter{
 		log:        log,
 		limit:      config.Limit,
 		reset:      config.Reset,
 		remaining:  config.Remaining,
 		earlyLimit: config.EarlyLimit,
 	}
+
+	if config.ClientLimits.Interval > 0 {
+		limiter.clientLimiter = rate.NewLimiter(rate.Limit(config.ClientLimits.Interval), config.ClientLimits.Requests)
```

Comment:
I may be misunderstanding what `rate.NewLimiter` expects but isn't the first argument the rate limit (requests / interval in seconds) and the second argument burstiness or size of the token bucket?